mypy/checker.py

```diff
@@ -2543,7 +2543,8 @@ def check_member_assignment(self, instance_type: Type, attribute_type: Type,
             [nodes.ARG_POS, nodes.ARG_POS], context)
 
         if not isinstance(inferred_dunder_set_type, CallableType):
-            self.fail(messages.DESCRIPTOR_SET_NOT_CALLABLE, context)
+            self.fail(messages.DESCRIPTOR_SET_NOT_CALLABLE
+                      .format(inferred_dunder_set_type), context)
```

Comment:
The code has two variations of this error message:

1. `"__set__ is not callable"` : used here
2. `"{}.__set__ is not callable"` : used elsewhere

I'd like to standardize on the second, but I'm not familiar enough with the code to know how to get the descriptor name here.  My naive attempt above did not trigger any test errors, but it should have, so I think it's not tested.  How should I proceed?

Reply:
What you are doing is not correct. I think you should use `attribute_type.type.name()` here, and add a test.

Reply:
This particular error is only generated in the case of an inferred `__set__`.  I'm not sure how to trigger this.  I tried a few variations along the lines of the tests in `check-inference.test` but all went through different code paths.  Any suggestions?  Is there a chance this is a dead code path?

Reply:
Yes, it looks like this is currently dead code: we only allow `__set__` to be a method above, and a method (i.e. function definition) can't have a non-callable type. I think this can be replaced with an assert.

Reply:
Addressed in my latest commit